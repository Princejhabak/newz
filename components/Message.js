import React from "react";
import { Alert } from 'react-bootstrap'

// Custom message component for the app to display and handle server errors
const Message = ({ variant, children }) => {
  return <Alert data-testid="alert" variant={variant}>{children}</Alert>
}

Message.defaultProps = {
  variant: 'info',
}

export default Message