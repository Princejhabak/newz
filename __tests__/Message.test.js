/* eslint-disable */

import React from 'react';
import { createRoot } from "react-dom/client";
import Message from "../components/Message";
import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { act } from 'react-dom/test-utils';

afterEach(cleanup);

test("renders without crashing", () => {
    const div = document.createElement("div");
    act(() => {
        createRoot(div).render(<Message></Message>);
    });
});


test("renders Message component correctly", () => {
    render(<Message>Error</Message>);
    expect(screen.getByTestId('alert')).toHaveTextContent("Error");
});

test("renders 2nd Message correctly", () => {
    render(<Message>Product not found</Message>);
    const messageElement = screen.getByTestId('alert');
    expect(messageElement).toBeInTheDocument();
    expect(messageElement).toHaveTextContent("Product not found");
});

test("matches snapshot", () => {
    const tree = renderer.create(<Message variant="danger">Error</Message>).toJSON();
    expect(tree).toMatchSnapshot();
});