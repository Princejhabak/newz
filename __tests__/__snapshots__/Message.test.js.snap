// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="fade alert alert-danger show"
  data-testid="alert"
  role="alert"
>
  Error
</div>
`;
